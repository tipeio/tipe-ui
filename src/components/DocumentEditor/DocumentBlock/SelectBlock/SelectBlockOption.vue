<template>
  <button
    :data-tipe-ui="$options.name"
    @click="$emit('click', label)"
  >
    <div class="layout">
      <div class="icon-container">
        <div class="icon-box">
          <tipe-icon
            :icon="icon"
            height="1.5rem"
            width="1.5rem"
          />
        </div>
      </div>
      <div class="label">{{ label }}</div>
      <div class="description">{{ description }}</div>
    </div>
  </button>
</template>

<script>
import blockOptionShape from '@/types/BlockOption'
import TipeIcon from '@/components/Icon'

export default {
  name: 'TipeSelectBlockOption',
  components: {
    TipeIcon
  },
  props: blockOptionShape
}
</script>

<style lang="postcss" scoped>
[data-tipe-ui='TipeSelectBlockOption'] {
  display: flex;
  flex-direction: row;
  height: 3.375rem;
  width: 100%;
  margin: 0;
  padding: 0;
  border: none;
  outline: none;
  background-color: transparent;
}

.layout {
  display: grid;
  grid-template-areas:
    'icon label'
    'icon description';
  grid-template-row: 1fr 1fr;
  grid-auto-columns: 3.375rem 1fr;
  height: 3.375rem;
}

.icon-container {
  grid-area: icon;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.icon-box {
  height: 2.625rem;
  width: 2.625rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(31, 52, 108, 0.05);
  border-radius: 0.25rem;
}

.label {
  grid-area: label;
  display: flex;
  align-items: flex-end;
  justify-content: flex-start;
  color: var(--dark-purple);
  font-size: 0.875rem;
  white-space: nowrap;
  overflow: hidden;
}

.description {
  grid-area: description;
  font-size: 0.75rem;
  color: var(--midPurple);
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  white-space: nowrap;
  overflow: hidden;
}
</style>
