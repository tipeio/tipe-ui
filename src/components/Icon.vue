<template>
  <div
    :class="size"
    :data-tipe-ui="$options.name"
  />
</template>

<script>
import vueTypes from 'vue-types'

const sizes = ['small', 'medium', 'large']

export default {
  name: 'TipeIcon',
  props: {
    size: vueTypes.oneOf(sizes).def('medium')
  }
}
</script>

<style scoped>
[data-tipe-ui='TipeIcon'] {
  background-color: red;
}

[data-tipe-ui='TipeIcon'].small {
  height: 1rem;
  width: 1rem;
}

[data-tipe-ui='TipeIcon'].medium {
  height: 2rem;
  width: 2rem;
}

[data-tipe-ui='TipeIcon'].large {
  height: 3rem;
  width: 3rem;
}
</style>
