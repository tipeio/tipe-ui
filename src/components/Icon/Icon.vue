<template>
  <object
    :data="src"
    type="image/svg+xml">
    <img :src="src" >
  </object>
</template>

<script>
import vueTypes from 'vue-types'

export default {
  name: 'TipeIcon',
  props: {
    icon: vueTypes.string
  },
  computed: {
    src() {
      return require(`../../icons/${this.icon}.svg`)
    }
  }
}
</script>
