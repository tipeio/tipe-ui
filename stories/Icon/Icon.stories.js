import { storiesOf } from '@storybook/vue'

import TipeIcon from '../../src/components/Icon'

const style = () => ({
  height: '1rem',
  width: '1rem'
})

const icons = [
  'check',
  'document',
  'folder',
  'key',
  'number',
  'share',
  'switch',
  'user',
  'color',
  'edit',
  'gear',
  'link',
  'pin',
  'sign-out',
  'text',
  'view',
  'copy',
  'flow',
  'grid',
  'list',
  'search',
  'star-active',
  'trash',
  'wrench',
  'credit-card',
  'folder-open',
  'group',
  'markdown',
  'media',
  'send',
  'star',
  'user-circle',
  'options',
  'plus',
  'cross',
  'arrow-down',
  'arrow-up',
  'arrow-right',
  'arrow-left'
]

storiesOf('Icon', module).add('icons', () => ({
  components: { TipeIcon },
  computed: { style },
  template:
    '<div>' +
    icons.reduce((template, icon) => {
      return (
        template +
        `<tipe-icon :style="style" icon="${icon}" color="rgb(255,0,0)" />`
      )
    }, '') +
    '</div>'
}))
