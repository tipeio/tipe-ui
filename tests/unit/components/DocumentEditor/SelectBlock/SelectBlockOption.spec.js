import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import SelectBlockOption from '@/components/DocumentEditor/DocumentBlock/SelectBlock/SelectBlockOption'
import Icon from '@/components/Icon'
import faker from 'faker'

describe('SelectBlockOption', () => {
  describe('<template>', () => {
    it('matches previous snapshot', () => {
      const renderer = createRenderer()
      const wrapper = shallowMount(SelectBlockOption)

      renderer.renderToString(wrapper.vm, (err, str) => {
        if (err) throw new Error(err)
        expect(str).toMatchSnapshot()
      })
    })

    it('has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(SelectBlockOption)

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeSelectBlockOption')
    })

    it('should render an Icon component', () => {
      const wrapper = shallowMount(SelectBlockOption)

      expect(wrapper.contains(Icon)).toEqual(true)
    })
  })

  describe(':props', () => {
    it(':icon - should pass icon prop to icon component', () => {
      const propsData = { icon: faker.lorem.word() }
      const wrapper = shallowMount(SelectBlockOption, { propsData })

      expect(wrapper.find(Icon).props().icon).toEqual(propsData.icon)
    })

    it(':label - should render label', () => {
      const propsData = { label: faker.lorem.word() }
      const wrapper = shallowMount(SelectBlockOption, { propsData })

      expect(wrapper.find('.label').text()).toEqual(propsData.label)
    })

    it(':label - should render description', () => {
      const propsData = { description: faker.lorem.word() }
      const wrapper = shallowMount(SelectBlockOption, { propsData })

      expect(wrapper.find('.description').text()).toEqual(propsData.description)
    })
  })

  describe('@events', () => {
    it('@click - should emit', () => {
      const wrapper = shallowMount(SelectBlockOption)
      wrapper.trigger('click')

      expect(wrapper.emitted().click).toBeTruthy()
    })
  })
})
