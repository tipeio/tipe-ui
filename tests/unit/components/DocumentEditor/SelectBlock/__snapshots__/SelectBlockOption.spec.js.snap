// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBlockOption <template> matches previous snapshot 1`] = `
<button data-tipe-ui="TipeSelectBlockOption">
  <div class="layout">
    <div class="icon-container">
      <div class="icon-box">
        <TipeIcon-stub></TipeIcon-stub>
      </div>
    </div>
    <div class="label"></div>
    <div class="description"></div>
  </div>
</button>
`;
